tools/cli/commands.js

```diff
@@ -261,49 +261,23 @@ export function parseRunTargets(targets) {
   });
 };
 
-const DEFAULT_MODERN = {
-    transpiler: true,
-    webArchOnly: true,
-    watcher: true,
-};
-
-const normalizeModern = (r = false) => Object.fromEntries(
-    Object.entries(DEFAULT_MODERN).map(([k, def]) => [
-        k,
-        r === true
-            ? def
-            : r === false || r?.[k] === false
-                ? false
-                : typeof r?.[k] === 'object'
-                    ? { ...r[k] }
-                    : def,
-    ]),
-);
-
-
-let modernForced = JSON.parse(process.env.METEOR_MODERN || "false");
-let meteorConfig;
-
-function getMeteorConfig(appDir) {
-  if (meteorConfig) return meteorConfig;
-  const packageJsonPath = files.pathJoin(appDir, 'package.json');
-  if (!files.exists(packageJsonPath)) {
-    return false;
+export function getMeteorConfig(appDir) {
```

Comment:
A more logical place for this might be the MeteorConfig class that is part of the project context: https://github.com/meteor/meteor/blob/2892718148a4453f31223b96e09aa7814ad80fae/tools/project-context.js#L1770

The context can be used in isobuild to provide the modern bundler config to the build plugins.

Reply:
I need the Meteor config loaded before project context instantiation so the watcher code can choose correctly (safe-watcher.ts). Maybe I can instantiate the config globally. Many parts are unclear and I’ve seen crashes in some areas as I change things around. I could give it a try on having more time.